core/consumer-rules.pro

```diff
@@ -0,0 +1,12 @@
+-dontwarn com.google.auto.value.AutoValue$Builder
```

Comment:
[Open question] Is there a way we can easily automate checking whether it's ok to discard these warnings, or whether we should instead keep these classes from getting modified/removed?

I ran some local tests by running the app and I didn't get any crashes due to missing classes, so I think it's ok to just use `dontwarn` for now, but I'm concerned that in the future we might actually have to keep classes instead. The only automated solution I can see now would be to create Android functional tests that would run in an emulator, though that's very slow to run in GH actions. However, if it's the only way, we should add them.

Reply:
I think reflection is the only problem, if auto value does reflection, better to have proper testing in place (I am not sure it does tho)

Reply:
Makes sense, thanks! AutoValue doesn't work with reflection so it seems like it should be fine.